launch/launch/actions/execute_process.py

```diff
@@ -183,13 +188,15 @@ def __init__(
         self.__cmd = [normalize_to_list_of_substitutions(x) for x in cmd]
         self.__name = name if name is None else normalize_to_list_of_substitutions(name)
         self.__cwd = cwd if cwd is None else normalize_to_list_of_substitutions(cwd)
-        self.__env = None  # type: Optional[List[Tuple[List[Substitution], List[Substitution]]]]
-        if env is not None:
-            self.__env = []
-            for key, value in env.items():
-                self.__env.append((
-                    normalize_to_list_of_substitutions(key),
-                    normalize_to_list_of_substitutions(value)))
+        if env is None:
+            env = dict(os.environ)
+        if additional_env is not None:
+            env.update(additional_env)
```

Comment:
@ivanpauno won't you be mutating `env` if it was user provided dictionary?

Reply:
https://github.com/ros2/launch_ros/issues/13#issuecomment-483319933